kornia/augmentation/_2d/intensity/random_snow.py

```diff
@@ -0,0 +1,67 @@
+from typing import Any, Dict, Optional, Tuple
+
+import torch
+
+from kornia.augmentation import random_generator as rg
+from kornia.augmentation._2d.intensity.base import IntensityAugmentationBase2D
+from kornia.color import hls_to_rgb, rgb_to_hls
+from kornia.core import Tensor
+from kornia.core.check import KORNIA_CHECK
+
+
+class RandomSnow(IntensityAugmentationBase2D):
+    r"""Generates snow effect on given tensor image or a batch tensor images.
+
+    Args:
+        snow_coefficient: A tuple of floats (lower and upper bound) between 0 and 1 that control
+        the amount of snow to add to the image, the larger value corresponds to the more snow.
+        brightness: A tuple of floats (lower and upper bound) greater than 1 that controls the
+        brightness of the snow.
+        same_on_batch: If True, apply the same transformation to each image in a batch. Default: False.
+        p: Probability of applying the transformation. Default: 0.5.
+        keepdim: Keep the output tensor with the same shape as input. Default: False.
+
+    Shape:
+        - Input: :math:`(C, H, W)` or :math:`(B, C, H, W)`
+        - Output: :math:`(B, C, H, W)`
+
+    Examples:
+        >>> inputs = torch.rand(2, 3, 4, 4)
+        >>> snow = kornia.augmentation.RandomSnow(p=1.0, snow_coefficient=(0.1, 0.6), brightness=(1, 5))
+        >>> output = snow(inputs)
+        >>> output.shape
+        torch.Size([2, 3, 4, 4])
+    """
+
+    def __init__(
+        self,
+        snow_coefficient: Tuple[float, float] = (0.5, 0.5),
+        brightness: Tuple[float, float] = (1.0, 1.0),
+        same_on_batch: bool = False,
+        p: float = 0.5,
+        keepdim: bool = False,
+    ) -> None:
+        super().__init__(p=p, same_on_batch=same_on_batch, keepdim=keepdim)
+        KORNIA_CHECK(all([0 <= el <= 1 for el in snow_coefficient]), "Snow coefficient values must be between 0 and 1.")
+        KORNIA_CHECK(all([1 <= el for el in brightness]), "Brightness values must be greater than 1.")
+
+        self._param_generator = rg.PlainUniformGenerator(
+            (snow_coefficient, "snow_coefficient", 0.5, (0.0, 1.0)), (brightness, "brightness", None, None)
+        )
+
+    def apply_transform(
+        self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any], transform: Optional[Tensor] = None
+    ) -> Tensor:
+        snow_coefficient = (params["snow_coefficient"].to(input)).mean()
```

Comment:
If you do re mean, does it really sample different for each element in the batch ?

Reply:
Ig no. But we need a snow_coefficient as float value and params dict stores tuples, as far as I understand, that's why mean is taken. I don't know if there's a way to make a snow_coefficient different for each element in the batch.

Reply:
I figured out how to do it, don't know if it's preferable.